src/MOIWrapper.jl

```diff
@@ -1,408 +1,435 @@
     else
-        error("Either row_lower or row_upper must be of abs less than 1e20")
+        Clp.set_maximum_seconds(model.inner, value)
     end
+    return nothing
 end
 
-function LQOI.delete_linear_constraints!(instance::Optimizer, start_row::Int, end_row::Int)
-    delete_rows(instance.inner, [Cint(i-1) for i in start_row:end_row])
-end
+# Clp behaves weird here
+MOI.get(model::Optimizer, ::MOI.TimeLimitSec) = Clp.maximum_seconds(model.inner)
+
+MOI.supports(::Optimizer, ::MOI.NumberOfThreads) = false
+
+
+# =====================
+# 
+# =======================
+MOI.supports_constraint(::Optimizer,
+    ::Type{MOI.ScalarAffineFunction{Float64}}, ::Type{S}
+) where{S <: SCALAR_SETS} = true
+
+MOI.supports_constraint(::Optimizer,
+    ::Type{MOI.SingleVariable}, ::Type{S}
+) where{S <: SCALAR_SETS} = true
+
+MOI.supports(::Optimizer, ::MOI.ObjectiveSense) = true
 
-function LQOI.change_linear_constraint_sense!(instance::Optimizer, rows::Vector{Int}, senses::Vector{Cchar})
-    lower = replace_inf.(get_row_lower(instance.inner))
-    upper = replace_inf.(get_row_upper(instance.inner))
-    for (sense, row) in zip(senses, rows)
-        lb = lower[row]
-        ub = upper[row]
-        if lb > -Inf
-            rhs = lb
-        elseif ub < Inf
-            rhs = ub
-        else
-            error("Either row_lower or row_upper must be of abs less than 1e20")
+MOI.supports(::Optimizer,
+    ::MOI.ObjectiveFunction{<:Union{
+        MOI.ScalarAffineFunction{Float64},
+        MOI.SingleVariable
+    }}
+) = true
+
+# =======================
+#   `copy_to` function
+# =======================
+
+function MOI.copy_to(
+    dest::Optimizer,
+    src::MOI.ModelLike;
+    copy_names=false
+)
+    # Check that all constraints and objectives
+    for (F, S) in MOI.get(src, MOI.ListOfConstraints())
+        MOI.supports_constraint(dest, F, S) || throw(MOI.UnsupportedConstraint{F, S}(
+            "Clp.Optimizer does not support constraints of type $F-in-$S."
+        ))
+    end
+    fobj_type = MOI.get(src, MOI.ObjectiveFunctionType())
+    if ! MOI.supports(dest, MOI.ObjectiveFunction{fobj_type}())
+        error("Unsupported objective $(fobj_type).")
+    end
+
+    # Empty dest
+    MOI.empty!(dest)
+
+    # Maps the indices of the 
+    mapping = MOIU.IndexMap()
+
+    # First create variables (including bounds)
+
+    # Create variables
+    # TODO: it may be more efficient to create the problem all at once
+    x_src = MOI.get(src, MOI.ListOfVariableIndices())
+    for (j, x) in enumerate(x_src)
+        # Variable j corresponds to j-th variable of x_src
+        mapping.varmap[x] = MOI.VariableIndex(j)
+        
+        # Possible bound combinations:
+        #=
+            * No bound (default case)
+            * Interval
+            * EqualTo
+            * LessThan
+            * GreaterThan
+            * LessThan && GreaterThan
+        =#
+        lb, ub = -Inf, Inf  # Default case: free variable
```

Comment:
Are we sure that `Inf` is correctly specified to Clp, and converted automatically to `DBL_MAX`?

Reply:
It does.
```julia
import Clp

m = Clp.ClpModel()
Clp.add_column(m, Cint(0), Int32[], Float64[], 0.0, Inf, 1.0)
Clp.get_col_upper(m)
```
returns
```julia
1-element Array{Float64,1}:
 1.7976931348623157e308
```